File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int rows) {
        
        vector<vector<int>>arr;
        arr.push_back({1});
        if(rows==1)
        return arr;

        for(int i=0; i<rows-1; i++)
        {
            vector<int>a;
            a.push_back(1);
            
            int x=0, y=1;
            
            while(y<arr[i].size()){
                a.push_back(arr[i][x]+arr[i][y]);
                x++;
                y++;
            }
            a.push_back(1);
            arr.push_back(a);
        }

        return arr;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        List<List<Integer>>list=new ArrayList<>();

        List<Integer>list1=new ArrayList<>();
        list1.add(1);
        list.add(list1);

        if(numRows==1)
        return list;

        int i=0;
        while(i<numRows-1){

             int x=0,y=1;
             List<Integer>arr=new ArrayList<>();
             arr.add(1);

             while(y<list.get(i).size()){
               arr.add(list.get(i).get(x)+list.get(i).get(y));
               x++;
               y++;
             }

             arr.add(1);
             list.add(arr);

          i++;
        }


         return list;
    }
}